crates/bevy_ecs/src/world/error.rs

```diff
@@ -1,12 +1,62 @@
-//! Contains error types returned by bevy's schedule.
+//! Contains error types returned by methods on [`World`].
+//!
+//! [`World`]: crate::world::World
+
+use std::any::TypeId;
 
 use thiserror::Error;
 
-use crate::schedule::InternedScheduleLabel;
+use crate::{component::ComponentId, prelude::*, schedule::InternedScheduleLabel};
 
 /// The error type returned by [`World::try_run_schedule`] if the provided schedule does not exist.
 ///
 /// [`World::try_run_schedule`]: crate::world::World::try_run_schedule
 #[derive(Error, Debug)]
 #[error("The schedule with the label {0:?} was not found.")]
 pub struct TryRunScheduleError(pub InternedScheduleLabel);
+
+/// The error type returned by [`World::get_reflect`] and [`World::get_reflect_mut`].
+///
+/// [`World::try_run_schedule`]: crate::world::World::try_run_schedule
+#[cfg(feature = "bevy_reflect")]
+#[derive(Error, Debug)]
+pub enum GetComponentReflectError {
+    /// There is no [`ComponentId`] corresponding to the given [`TypeId`].
+    ///
+    /// Did you call [`App::register_type()`]?
```

Comment:
Nit: I _love_ that you have doc comments on these error variants (a lot of people forget to add them)! My one suggestion would be that they should probably be more doc-reader-facing rather than error-message-facing.

For example, instead of `Did you call [`App::register_type()`]?` you could explain that this error often occurs because the user forgot to register the type with `App::register_type`.

Same for the `MissingReflectFromPtrTypeData` variant.

Reply:
Ah, good point! I actually noticed that as well while writing the docs, but think I needed a break at that point and just left it as-is :see_no_evil:

But yeah, I agree, and fixed it now :)